flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/InputGateUtil.java

```diff
@@ -38,18 +40,18 @@ public static InputGate createInputGate(Collection<IndexedInputGate> inputGates1
 		List<IndexedInputGate> gates = new ArrayList<>(inputGates1.size() + inputGates2.size());
 		gates.addAll(inputGates1);
 		gates.addAll(inputGates2);
-		return createInputGate(gates.toArray(new IndexedInputGate[gates.size()]));
+		return createInputGate(gates);
 	}
 
-	public static InputGate createInputGate(IndexedInputGate[] inputGates) {
-		if (inputGates.length <= 0) {
+	public static InputGate createInputGate(Collection<IndexedInputGate> inputGates) {
```

Comment:
Replacing an array with `Collection` allows to lose ordering. Why not to use `List` instead?

Reply:
I had to change a bit of the invoking code, but it's not much because ultimately Lists are used everywhere.